### ISSUE TITLE
Need to detect artifact

### ISSUE THREAD

**Issue by mapo80** [Guillem96/deep-erase#3]
Hi,

I found this project very interesting.
I need to detect if an image contains ink artifacts, I don't need to cleaned images.
According to you can I use this model to find all artifacts contained on images?

Thanks

**Comment by jponf** [comment]
Hi @mapo80 , 

The model is designed to remove ink artifacts. For detection I would recommend adapting the data generator to generate "heat-maps", similar to what they do in CRAFT: https://arxiv.org/abs/1904.01941, and train a single U-net to generate those heat-maps.

Then, you can compute the polygons that enclose those heat-maps with classical computer vision algorithms. This model should be able to handle any number of artifacts in an image, provided that it has been trained properly.

The paper is quite easy to read and we used this approach for something different in the past and it worked quite well.

@Guillem96  do you have any suggestion?

**Comment by mapo80** [comment]
Hi @jponf, I don't how to thank you. Thanks very much for all your support. I'll give it a try.
Thanks very much!

**Comment by mapo80** [comment]
Hi, I want to send some example of artifacts that I need to detect. According to you, could it works?

![202402121016348500047](https://github.com/Guillem96/deep-erase/assets/1677958/66fe6321-842d-476a-8745-c1400d6ac266)

![202402121016348500019](https://github.com/Guillem96/deep-erase/assets/1677958/6e40c176-2daa-4118-bbbc-7b2123c93a46)

![202402121016348500022](https://github.com/Guillem96/deep-erase/assets/1677958/14378718-d99c-4bb9-a60a-a8573947a930)

**Comment by jponf** [comment]
Yes, it should work with these samples. I've been trying to find the code that we had to generate the "heat-map" for training, but it's probably long gone from my archives.

The idea is simple though, you keep clean images and artifact images as two inputs to your data generator. Then, simply blend the artifacts with the clean image. To get even more variability, you can apply transformations to the artifacts.

If your artifacts have weird shapes and it's hard to generate the heat-map programmatically you can always create it on an image manipulation program (i.e, GIMP). Then your data generator simply applies the same transformations to the artifact sample and the accompanying heat-map sample to generate the (input, output) pairs.

For more information I suggest looking at the original CRAFT repository: https://github.com/clovaai/CRAFT-pytorch/tree/master

**Comment by mapo80** [comment]
Ok thanks very much. I'll give it a try!

Thanks
